### PR TITLE
Add GET graffiti keymanager api

### PR DESCRIPTION
**Pull request by courtneyeh**
#7690 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/GetGraffiti.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    final BLSPublicKey publicKey = request.getPathParameter(PARAM_PUBKEY_TYPE);
+
+    final Optional<Validator> maybeValidator = keyManager.getValidatorByPublicKey(publicKey);
+    if (maybeValidator.isEmpty()) {
+      request.respondError(SC_NOT_FOUND, "Validator not found");
+      return;
+    }
+
+    String graffiti = maybeValidator.get().getGraffiti().map(this::processGraffitiBytes).orElse("");
```
Comment (rolfyone) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/GetGraffiti.java]: final

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/GetGraffiti.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        return false;
+      }
+      GraffitiResponse that = (GraffitiResponse) o;
+      return Objects.equals(publicKey, that.publicKey) && Objects.equals(graffiti, that.graffiti);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(publicKey, graffiti);
+    }
```
Comment (rolfyone) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/GetGraffiti.java:102]: not sure we need these?

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/GetGraffiti.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    String graffiti = maybeValidator.get().getGraffiti().map(this::processGraffitiBytes).orElse("");
+    request.respondOk(new GraffitiResponse(publicKey, graffiti));
+  }
+
+  private String processGraffitiBytes(final Bytes32 graffiti) {
+    return new String(graffiti.toArrayUnsafe(), StandardCharsets.UTF_8).strip().replace("\0", "");
+  }
+
+  static class GraffitiResponse {
+    private final Optional<String> publicKey;
```
Comment (rolfyone) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/GetGraffiti.java]: generally speaking we'd be having Optional<Bytes32> here i think in most objects, not sure why we're going the string route. String at rendering is fine but we lose information by making this string here.

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/GetGraffiti.java (hunk 2 above)
Comment (courtneyeh) [reply validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/GetGraffiti.java:102]: Did for the testing, comparing the output

File: validator/client/src/test/resources/tech/pegasys/teku/validator/client/restapi/schema/GraffitiResponse.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  "type" : "object",
+  "required" : [ "data" ],
+  "properties" : {
+    "data" : {
+      "type" : "object",
+      "required" : [ "graffiti" ],
+      "properties" : {
+        "pubkey" : {
+          "type" : "string"
+        },
```
Comment (rolfyone) [inline validator/client/src/test/resources/tech/pegasys/teku/validator/client/restapi/schema/GraffitiResponse.json]: pubkey isn't resolved.

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/GetGraffiti.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            .summary("Get Graffiti")
+            .description(
+                "Get the graffiti for an individual validator. If no graffiti is set explicitly, returns the process-wide default.")
+            .tags(TAG_GRAFFITI)
+            .withBearerAuthSecurity()
+            .pathParam(PARAM_PUBKEY_TYPE)
+            .response(SC_OK, "Success response", RESPONSE_TYPE)
+            .withAuthenticationResponses()
+            .withNotFoundResponse()
+            .withNotImplementedResponse()
```
Comment (rolfyone) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/GetGraffiti.java:61]: we lost any tests that would show basic metadata functionality, weren't they present before? if not they need adding.

File: validator/client/src/test/java/tech/pegasys/teku/validator/client/restapi/apis/GetGraffitiTest.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+class GetGraffitiTest {
+  private final GetGraffiti handler = new GetGraffiti();
+
+  private final DataStructureUtil dataStructureUtil =
+      new DataStructureUtil(TestSpecFactory.createDefault());
+
+  @Test
+  void metadata_shouldHandle200() throws JsonProcessingException {
+    GetGraffiti.GraffitiResponse response =
```
Comment (rolfyone) [inline validator/client/src/test/java/tech/pegasys/teku/validator/client/restapi/apis/GetGraffitiTest.java]: final

**Review by lucassaldanha — APPROVED** [review]
LGMT

**Pull request merged**